source/backend/api/Areas/CompensationRequisition/Controllers/CompensationRequisitionController.cs

```diff
@@ -267,5 +267,71 @@ public IActionResult GetCompensationRequisitionLeasePayees([FromRoute] long id)
 
             return new JsonResult(_mapper.Map<IEnumerable<CompReqLeasePayeeModel>>(compReqPayees));
         }
+
+        [HttpGet("{id:long}/historical")]
+        [HasPermission(Permissions.CompensationRequisitionView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(List<CompensationRequisitionModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "compensation-requisition" })]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        public IActionResult GetCompensationRequisionAtTime([FromRoute] long id, [FromQuery] DateTime time)
+        {
+
+            var histCompReq = _compensationRequisitionService.GetCompensationRequisitionAtTime(id, time);
```

Comment:
hmm, for compensation we did use the service instead of the repository? not sure why.

Reply:
The comp req service is doing additional composition to the entity.